admin/admin-base.php

```diff
@@ -266,27 +266,31 @@ public function admin_print_footer_scripts() {
 		<script type="text/javascript">
 			if (typeof jQuery != 'undefined') {
 				jQuery(
-					function($){
-						$.ajaxPrefilter(function (options, originalOptions, jqXHR) {
+					function( $ ){
+						$.ajaxPrefilter( function ( options, originalOptions, jqXHR ) {
 							if ( -1 != options.url.indexOf( ajaxurl ) || -1 != ajaxurl.indexOf( options.url ) ) {
-								if ( 'undefined' === typeof options.data ) {
-									options.data = ( 'get' === options.type.toLowerCase() ) ? '<?php echo $str; // phpcs:ignore WordPress.Security.EscapeOutput ?>' : <?php echo $arr; // phpcs:ignore WordPress.Security.EscapeOutput ?>;
+
+								function addStringParamaters() {
+									if ( 'undefined' === typeof options.data || '' === options.data ) {
+										options.data = '<?php echo $str; // phpcs:ignore WordPress.Security.EscapeOutput ?>';
+									} else {
+										options.data = options.data + '&<?php echo $str; // phpcs:ignore WordPress.Security.EscapeOutput ?>';
+									}
+								}
+
+								// options.processData set to true is the default jQuery process where the data is converted in a query string by using jQuery.param().
+								// This step is done before applying filters. Thus here the options.data is already a string in this case.
+								// @See https://github.com/jquery/jquery/blob/3.5.1/src/ajax.js#L563-L569 jQuery ajax function.
+								if ( options.processData ) {
+									addStringParamaters();
 								} else {
-									if ( 'string' === typeof options.data ) {
-										if ( '' === options.data && 'get' === options.type.toLowerCase() ) {
-											options.url = options.url+'&<?php echo $str; // phpcs:ignore WordPress.Security.EscapeOutput ?>';
-										} else {
-											try {
-												var o = JSON.parse(options.data);
-												o = $.extend(o, <?php echo $arr; // phpcs:ignore WordPress.Security.EscapeOutput ?>);
-												options.data = JSON.stringify(o);
-											}
-											catch(e) {
-												options.data = '<?php echo $str; // phpcs:ignore WordPress.Security.EscapeOutput ?>&'+options.data;
-											}
-										}
+									// If options.processData is set to false data could be undefined or pass as a string.
+									// So data as to be processed as if options.processData is set to true.
+									if ( 'undefined' === typeof options.data || 'string' === typeof options.data ) {
```

Comment:
What would you think to combine all the tests in one?

```php
if ( options.processData || 'undefined' === typeof options.data || 'string' === typeof options.data ) {
addStringParameters();
} else {
options.data = Object.assign( options.data, <?php echo $arr; // phpcs:ignore WordPress.Security.EscapeOutput ?> );
}
```

It would even allow us to remove the function `addStringParameters()` if we want to reduce the js size, as it woud be called only once (maybe with a price on the readability though).

May I suggest to use the syntax` /* */` for multiline comments as in https://github.com/polylang/polylang/blob/3.0-beta1/include/class-polylang.php#L55-L58. I know we have more comments with the single line syntax in Polylang but I started to be more consistent in the recent past and use single line syntax for single line comments and multiline syntax for multiline comments.

Reply:
Ok for multiline comments.
As we discussed in live together, it seems preferable for readibility - correctly understand each case - to keep the code as this.